### PR TITLE
Jetpack Section: add backup download notice

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #15191
Related PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/344

### To test

1. Open Activity List or Backups screen
2. Trigger a backup
3. After the backup is finished you should see a notice at the top to download the file
4. Tap Download, you should be redirected to the browser with the option to download the file
5. Go back to the app, tap dismiss
6. Refresh the list, the notice should not appear again

(You can also dismiss on Calypso and check in the app, both platforms are synced)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by leandroalonso** [comment]
@osullivanchris JFYI: I used the same "component" used in the Notifications. We might want to change that in the future but as for now, I think it works ok. What are your thoughts?

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 1]
```diff
@@ -231,6 +235,33 @@ class ActivityListViewModel: Observable {
...
+        guard let validUntil = store.getBackupStatus(site: site)?.validUntil,
+              Date() < validUntil,
+              let backupPoint = store.getBackupStatus(site: site)?.backupPoint,
+              let downloadURLString = store.getBackupStatus(site: site)?.url,
+              let downloadURL = URL(string: downloadURLString),
+              let downloadID = store.getBackupStatus(site: site)?.downloadID else {
+            return nil
+        }
+
+        downloadPromptView.setupHeading("We successfuly created a backup of your site as of \(longDateFormatterWithTime.string(from: backupPoint))")
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: `NSLocalizedString` please?

```suggestion
        downloadPromptView.setupHeading("We successfully created a backup of your site as of \(longDateFormatterWithTime.string(from: backupPoint))")
```

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 2]
```diff
@@ -231,6 +235,33 @@ class ActivityListViewModel: Observable {
...
+
+        downloadPromptView.setupYesButton(title: "Download") { _ in
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: `NSLocalizedString` please?

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 3]
```diff
@@ -231,6 +235,33 @@ class ActivityListViewModel: Observable {
...
+            UIApplication.shared.open(downloadURL)
+        }
+
+        downloadPromptView.setupNoButton(title: "Dismiss") { [weak self] button in
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: `NSLocalizedString` please?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Awesome, looking good, left a couple comments and have a couple questions:

- Should the background in light mode be white? 
- If I wait for the download to be created then dismiss that view I have to manually pull to refresh to get the status to appear. Should we auto refresh?

**Comment by osullivanchris** [comment]
> @osullivanchris JFYI: I used the same "component" used in the Notifications. We might want to change that in the future but as for now, I think it works ok. What are your thoughts?

👋  Hey! I didn't know that you had committed to doing this. Great stuff! Its been on my list this week to explore it when I can get around to it.

Yeah I totally agree, this looks great for now. I'd like to see light mode too as @emilylaguna mentioned it. A screenshot would be fine. In dark mode I kinda expected to see a background on that area too. But I don't know if the component you're taking from notifications has a background or not.

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: Done in 7126c1450e2dbcdce6553a94715d0a8e9a9f32e1

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: Done in 7126c1450e2dbcdce6553a94715d0a8e9a9f32e1

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: Done in 7126c1450e2dbcdce6553a94715d0a8e9a9f32e1

**Comment by leandroalonso** [comment]
@emilylaguna 

> Should the background in light mode be white?

I reviewed with @osullivanchris and as for now, we're keeping it as it is.

> If I wait for the download to be created then dismiss that view I have to manually pull to refresh to get the status to appear. Should we auto refresh?


Done in dbae49caec09ad98f317de2740ad194e278515d0

**Review by emilylaguna — APPROVED** [review]
LGTM! 🚀

**Pull request merged**